src/main/java/sx/blah/discord/api/internal/DiscordWS.java

```diff
@@ -118,7 +118,7 @@ public void onWebSocketClose(int statusCode, String reason) {
 		Discord4J.LOGGER.info(LogMarkers.WEBSOCKET, "Shard {} websocket disconnected with status code {} and reason \"{}\".", shard.getInfo()[0], statusCode, reason);
 
 		heartbeatHandler.shutdown();
-		if (this.state != State.DISCONNECTING && statusCode != 4003 && statusCode != 4004 && statusCode != 4005 && statusCode != 4010) {
+		if (this.state != State.DISCONNECTING && statusCode != 4003 && statusCode != 4004 && statusCode != 4005 && statusCode != 4010 && statusCode == 1001 ^ reason.equals("Shutdown")) {
```

Comment:
You should put parenthesis around `statusCode == 1001 ^ reason.equals("Shutdown")` just for clarity in case we refactor this